src/main/java/com/textmessenger/service/UserServiceImpl.java

```diff
@@ -2,36 +2,81 @@
 
 import com.textmessenger.model.entity.Notification;
 import com.textmessenger.model.entity.Post;
+import com.textmessenger.model.entity.TemporaryToken;
 import com.textmessenger.model.entity.User;
 import com.textmessenger.model.entity.dto.UserToFrontShort;
+import com.textmessenger.repository.TemporaryTokenRepository;
 import com.textmessenger.repository.UserRepository;
 import com.textmessenger.security.UserPrincipal;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.mail.SimpleMailMessage;
 import org.springframework.security.core.context.SecurityContextHolder;
 import org.springframework.security.core.userdetails.UsernameNotFoundException;
 import org.springframework.security.crypto.password.PasswordEncoder;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
 
+import java.util.Date;
 import java.util.List;
 import java.util.Optional;
+import java.util.UUID;
 
 @Service
 @Transactional
 public class UserServiceImpl implements UserService {
   @Autowired
   PasswordEncoder passwordEncoder;
   private final UserRepository userRepository;
+  private final TemporaryTokenRepository temporaryTokenRepository;
   private UserToFrontShort userToFront;
+  private final EmailService emailService;
 
-  public UserServiceImpl(UserRepository userRepository) {
+
+  public UserServiceImpl(UserRepository userRepository, TemporaryTokenRepository temporaryTokenRepository,
+                         EmailService emailService) {
     this.userRepository = userRepository;
+    this.temporaryTokenRepository = temporaryTokenRepository;
+    this.emailService = emailService;
+  }
+
+  @Override
+  public String setUserIsEnabled(String token) {
+    Optional<TemporaryToken> byToken = temporaryTokenRepository.findByToken(token);
+    if (byToken.isPresent()) {
+      User user = byToken.get().getUser();
+      if (byToken.get().getExpiryDate().compareTo(new Date()) <= 0) {
+        user.setEnabled(true);
+        userRepository.save(user);
+        SimpleMailMessage email = new SimpleMailMessage();
+        email.setTo(user.getEmail());
+        email.setSubject("Congratulation your account is activate");
+        email.setText("Enjoy our application");
+        emailService.sendEmail(email);
+        temporaryTokenRepository.delete(byToken.get());
+        return new String("your user is activate");
+      }
+      TemporaryToken temporaryToken = byToken.get();
+      temporaryToken.setToken(UUID.randomUUID().toString());
+      temporaryToken.setExpiryDate(temporaryToken.calculateExpiryDate(new Date().getMinutes()));
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this use of "getMinutes"; it is deprecated. ([squid:CallToDeprecatedMethod](https://sonarcloud.io/organizations/final_project_text_messenger/rules#rule_key=squid%3ACallToDeprecatedMethod))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=Text_Messenger_Project&pullRequest=166&issues=AWWhHkYc3Qu1svVwBtp8&open=AWWhHkYc3Qu1svVwBtp8)

[ISSUE_KEY]: AWWhHkYc3Qu1svVwBtp8